release/src/router/rc/wanduck.c

```diff
@@ -2352,10 +2352,11 @@ void handle_dns_req(int sfd, unsigned char *request, int maxlen, struct sockaddr
 		return;
 	opcode = d_req->flag_set.flag_num & htons(0x7800);
 	ptr = request + sizeof(dns_header);
-	end = request + maxlen;
+	/* queries.name is PATHLEN long, one trailing byte for null */
+	end = request + (maxlen > PATHLEN-1 ? PATHLEN-1 : maxlen);
```

Comment:
unrelated, dns labels don't use trailing null bytes. end pointer is the absulute end of the packet, not of single hostname or so.

Reply:
This has less to do with DNS but more with the fact that strcat() strcpy() calls rely on the position of the null byte for their operations.

Reply:
sure, you’re right, but I mean a bit different thing. cutting “end” too early at pathlen-1 would cut off type/class fields placed after dns name, therefore it will efficiently break validation at all

Reply:
Ah, got it and you're right. I'd usually make it buffer[PATHLEN+1] but I didn't want to modify the .h file. But again you're right about breaking DNS parsing in this case.